api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -366,13 +371,30 @@ public TableResult previewBigQueryJobConfig(DataSetPreviewRequest request) {
 
     if (supportsConceptSets(domain)) {
       List<DbConceptSetConceptId> dbConceptSetConceptIds;
+      List<Long> dbCriteriaAnswerIds = new ArrayList<>();
       switch (domain) {
         case SURVEY:
         case PHYSICAL_MEASUREMENT_CSS:
           dbConceptSetConceptIds =
               isPrepackagedAllSurveys(request)
                   ? findPrepackagedSurveyQuestionConceptIds()
                   : findDomainConceptIds(request.getDomain(), request.getConceptSetIds());
+          List<Long> questionConceptIds =
+              dbConceptSetConceptIds.stream()
+                  .map(DbConceptSetConceptId::getConceptId)
+                  .collect(Collectors.toList());
+          // find any questions that belong to PFHH survey. The PFHH survey should
+          // only use answer ids when looking up participants.
+          List<Long> pfhhSurveyQuestionIds = findPFHHSurveyQuestionIds(questionConceptIds);
+          if (!pfhhSurveyQuestionIds.isEmpty()) {
+            // need to filter out PFHH survey questions for other survey questions
+            dbConceptSetConceptIds =
+                dbConceptSetConceptIds.stream()
+                    .filter(cid -> !pfhhSurveyQuestionIds.contains(cid.getConceptId()))
+                    .collect(Collectors.toList());
+            // find all answers for the questions
+            dbCriteriaAnswerIds = findPFHHSurveyAnswerIds(pfhhSurveyQuestionIds);
+          }
```

Comment:
Why is this block under `case PHYSICAL_MEASUREMENT_CSS:` and not under `case SURVEY:` or under default?
I do not understand

Reply:
Notice the code, both PHYSICAL_MEASUREMENT_CSS and SURVEY fall under this section:
![Screen Shot 2022-11-30 at 9 39 43 AM](https://user-images.githubusercontent.com/3904919/204842231-fae9bc4a-7d8e-443f-9847-0edb0fa56c6d.png)